cloud/new_ec2.go

```diff
@@ -548,8 +548,63 @@ func (m *ec2Manager) IsUp(h *host.Host) (bool, error) {
 }
 
 // OnUp is called when the host is up.
-func (m *ec2Manager) OnUp(*host.Host) error {
-	// Unused since we set tags in SpawnHost()
+func (m *ec2Manager) OnUp(h *host.Host) error {
+	if isHostSpot(h) {
```

Comment:
shouldn't we tag on demand hosts the same way? I think its right to attach tags to on-demand hosts elsewhere, we should at least call common code...

Reply:
We tag on-demand hosts when we spawn them, but "spawning" a spot host doesn't actually spawn a host, so we have to wait until OnUp. Separately, the code to tag a spot instance after launch is mostly different from the code to tag resources. For an on-demand host, we use the instance ID returned by the RunInstances call. For a spot host, we have to call DescribeSpotInstanceRequests and then DescribeInstances. Although you might expect there to be shared code, there is very little of it. I think this is correct the way it is written.